backend/src/server.ts

```diff
@@ -647,16 +647,82 @@ app.delete('/spotlight/provider_keys/:keyUuid', async (request: AuthenticatedReq
 // Spotlight - Model Proxy (Updated Architecture)
 // ============================================================================
 
-app.post('/spotlight/:provider/v1/messages', async (request: FastifyRequest, reply: FastifyReply) => {
+// Catch-all route for all provider API endpoints (not just /v1/messages)
```

Comment:
Remove extra comments throughout the file like this. We should aim for minimal comments only in places where the code itself is complex or unintuitive.